src/cpp-common/include/cpp-common/HashMap.tpp

```diff
@@ -0,0 +1,145 @@
+#ifndef __HASH_MAP_TPP_
+#define __HASH_MAP_TPP_
+
+#include "HashMap.h"
+#include <cstdlib>
+#include <cstring>
+#include <new>
+#include <string>
+
+template <typename Key, typename MappedType, uint16_t maxSize>
+HashMap<Key, MappedType, maxSize>::HashMap() : m_usedSpaces(0) {
+    memset(m_usedSpacesFlag, 0, sizeof(m_usedSpacesFlag));
+}
+template <typename Key, typename MappedType, uint16_t maxSize>
+HashMap<Key, MappedType, maxSize>::~HashMap() {
+    // clear();
+}
+
+template <typename Key, typename MappedType, uint16_t maxSize>
+bool HashMap<Key, MappedType, maxSize>::insert(const std::pair<Key, MappedType>& item) {
+    bool loopedOnce = false;
+    uint16_t index = hash(item.first);
+    do {
+        if (m_usedSpacesFlag[index] == false) {
+            new (&m_storage[index]) std::pair<Key, MappedType>(item);
+            m_usedSpaces++;
+            m_usedSpacesFlag[index] = true;
+            return true;
+        }
+        if (m_usedSpacesFlag[index] == true) {
+            // If a key is already present in the map, do not overwrite its value
```

Comment:
Is this really the behavior that we want? If we want to overwrite a value we need to remove it, then insert. Or we could have another function, one that overwrites, one that does overwrite. In general insert overwrites though

Reply:
i'll overwrite, will unify logic in a single if

Reply:
Not sure i understand, what I meant was instead of doing this
```cpp
hash.insert(1, obj1);
hash.insert(1, obj2); // Overwriting
```

i need to do. 
```cpp
hash.insert(1, obj1);
hash.remove(1);
hash.insert(1, obj2); // Overwriting
```

I just want to make sure that's what we want.